### ISSUE TITLE
OnlyOffice won't load via a Shared Link to a file in NC 16, but works fine in NC 15

### ISSUE THREAD

**Issue by smcserv** [ONLYOFFICE/onlyoffice-nextcloud#88]

**Comment by smcserv** [comment]
I have both NC 15.0.7 and NC 16.0.1 running in two separate VM within the same subnet in addition to OnlyOffice DS 5.3.0-243 in another VM. All VMs are running Ubuntu 18.04.2 LTS.

In NC 16, when I shared a file through a Public link (with or without "Edit" permission), the link will open NextCloud with the option to Download the file, but NOT the option to Open the File in OnlyOffice for viewing or editing. The same process works fine in NC 15.

I get the following Error Message in NC's Log:

{"reqId":"TBxQu4sQGtDY62KBCShd","level":3,"time":"2019-05-31T01:29:21+00:00","remoteAddr":"207.6.xxx.xxx","user":"--","app":"jsresourceloader","method":"GET","url":"\/s\/sJRRZcZsLT2ps9T","message":"Could not find resource core\/vendor\/marked\/marked.min.js to load","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Ubuntu Chromium\/74.0.3729.169 Chrome\/74.0.3729.169 Safari\/537.36","version":"16.0.1.1"}

Upon further examination, I noticed the referenced JS component "marked.min.js" no longer exist in NC 16, but is located in the expected referenced location in NC 15!

This bug is causing OnlyOffice to be NOT loaded if accessed via a shared public link.

I have not yet tested the same scenario when the link is shared with other NC users. I will update this post once I have a chance to test.

**Comment by alexanderonlyoffice** [comment]
Hello @smcserv, unfortunately, we cannot reproduce the issue on our side. Did this issue occur before the updating NC from version 15 to 16 on this exact machine?

**Comment by smcserv** [comment]
Hi @alexanderonlyoffice, the NC 16 VM was a brand new fresh install from scratch with Ubuntu Server 18.04.02 LTS, PHP 7.2, Apache 2.4.29, Redis v5.0.3. The  NC 15 VM was also a fresh install with identical setup except for NC15 which was installed with v15.0.4 and upgraded through to v15.0.07.

Both VMs are running with identical set of Apps from NextCloud AppStore.

As I've mentioned before, the NC15 VM has the proper "marked.min.js" file in its "./core/vendor/marked/" folder, whereas for NC16, no such file can be found! 

I have both OnlyOffice Document Server and Collabora CODE setup on two other separate VM on the same subnet as the NC15 & 16 VMs. I only enabled either the "OnlyOffice" or "Collabora Online" NextCloud plugin one at time, not both at the same time. The "Collabora Online" plugin works perfectly with no issues, but the "OnlyOffice" plugin has the issue mentioned in this post.

Again, on the NC15 VM, everything works fine. But on the NC16 VM, only the "OnlyOffice" plugin generates the error when trying to open a Publicly Shared Link. 

Please let me know what other info you would need to track down this issue.

Much appreciated for you help...Thanks!

**Comment by smcserv** [comment]
Hi @alexanderonlyoffice,

Problem **SOLVED!**

Sorry for sending you on a wild goose chase!

I should have been more diligent in testing and tracking down the Issue in the first place. 

The posted Error Message did led me to think there's something coming from a source related to "Markdown" editing or viewing that might be causing the issue! And sure enough, I had ONE extra App enabled in NC 16 that I did not have in NC 15, the "README.md" App from NextCloud AppStore!

This App was generating the Error Message and preventing the OnlyOffice App from loading properly, somehow. Anyway, after I disabled this "README.md" App, this post's issue was gone. NextCloud 16 now offers the option to load OnlyOffice to view/edit the shared document when shared via a Publicly Shared Link!

Anyway, Thank You for initially looking into this issue in the 1st place!
